src/Facets/SynapseBridgeFacet.sol

```diff
@@ -0,0 +1,126 @@
+        0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    /// @notice The contract address of the SynapseRouter on the source chain.
+    ISynapseRouter public immutable synapseRouter;
+
+    /// Types ///
+
+    /// @param originQuery Origin swap query. Empty struct indicates no swap is required.
+    /// @param destQuery Destination swap query. Empty struct indicates no swap is required.
+    struct SynapseData {
+        ISynapseRouter.SwapQuery originQuery;
+        ISynapseRouter.SwapQuery destQuery;
+    }
+
+    /// Constructor ///
+
+    /// @notice Initialize the contract.
+    /// @param _synapseRouter The contract address of the SynapseRouter on the source chain.
+    constructor(ISynapseRouter _synapseRouter) {
+        synapseRouter = _synapseRouter;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Synapse Bridge
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _synapseData data specific to Synapse Bridge
+    function startBridgeTokensViaSynapseBridge(
+        ILiFi.BridgeData calldata _bridgeData,
+        SynapseData calldata _synapseData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+
+        _startBridge(_bridgeData, _synapseData);
+    }
+
+    /// @notice Performs a swap before bridging via Synapse Bridge
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _swapData an array of swap related data for performing swaps before bridging
+    /// @param _synapseData data specific to Synapse Bridge
+    function swapAndStartBridgeTokensViaSynapseBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        SynapseData calldata _synapseData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+
+        _startBridge(_bridgeData, _synapseData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via Synapse Bridge
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _synapseData data specific to Synapse Bridge
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        SynapseData calldata _synapseData
+    ) internal {
+        uint256 nativeAssetAmount;
```

Comment:
I dont really see the added benefit by this variable. Could also just use _bridgeData.minAmount directly if the value is not manipulated.

Reply:
you mean we can check if the sending asset is native asset again when we call `bridge` function?

Reply:
ahh... .sorry, I misread the code. Forget my comment, please :)